src/core/segment_allocator.h

```diff
@@ -17,16 +17,19 @@ namespace dfly {
 
 /**
  * @brief Tightly coupled with mi_malloc 2.x implementation.
- *        Fetches 4MiB segment pointers from the allocated pointers.
+ *        Fetches 32MiB segment pointers from the allocated pointers.
```

Comment:
It doesn't really make a difference if we use `8MiB` here as it was before we switched to mimalloc version that used `4MiBSegments` because either way `SmallString` will hold a `4 bytes` pointer so it won't affect its memory footprint which is the goal of this anyway